flow/raster_cache.cc

```diff
@@ -115,7 +117,11 @@ sk_sp<SkImage> RasterCache::GetPrerolledImage(GrContext* context,
         (is_complex || isWorthRasterizing(picture))) {
       TRACE_EVENT2("flutter", "Rasterize picture layer", "width",
                    physical_size.width(), "height", physical_size.height());
-      SkImageInfo info = SkImageInfo::MakeN32Premul(physical_size);
+      SkImageInfo info = SkImageInfo::Make(physical_size.width(),
+                                           physical_size.height(),
+                                           kRGBA_8888_SkColorType,
+                                           kPremul_SkAlphaType,
+                                           sk_ref_sp(dst_color_space));
```

Comment:
Just for my reference, what is the convention followed in Skia w.r.t. subclasses of `SkRefCnt`. Do you always make it a point to hide constructors and provide factories that provide `sk_sp` instances of the same? I have been confused about this in the past. I created an `SkRefCnt` object on the stack and have passed it to a method that took a raw pointer assuming that meant I controlled the lifetime of the object. Unfortunately, that object would then be referenced via `sk_ref_sp` and cause double-frees when it went out of scope.

If the convention is to always use `sk_sp`, maybe the frame context should take a `sk_sp<SkColorSpace>` instead of a raw pointer.